addons/account/models/account_move_line.py

```diff
@@ -1353,7 +1353,16 @@ def _sanitize_vals(self, vals):
                 vals.pop('debit', None)
                 vals.pop('credit', None)
             else:
-                vals['balance'] = vals.pop('debit', 0) - vals.pop('credit', 0)
+                move = self.env['account.move']
+                if 'move_id' in vals:
+                    move = self.env['account.move'].browse(vals['move_id'])
+                if move and not move.is_invoice(include_receipts=True) and vals.get('debit') is False and vals.get('credit') is False:
```

Comment:
we should also check if the valance is in the vals because if it is, we probably don't want to override them